Building on MacOS

Mac users may not be able to build Rimmel, due to this error:

```sh
Users/username/rimmel/node_modules/rollup/dist/native.js:59
        throw new Error(
        ^
Error: Cannot find module @rollup/rollup-darwin-arm64. npm has a bug related to optional dependencies<a href="https://github.com/npm/cli/issues/4828" target="_blank" rel="noopener noreferrer nofollow"></a>. Please try npm i again after removing both package-lock.json and node_modules directory.
    at requireWithFriendlyError (/Users/username/rimmel/node_modules/rollup/dist/native.js:59:9)
    at Object.<anonymous> (/Users/username/rimmel/node_modules/rollup/dist/native.js:68:76)
    at Module._compile (node:internal/modules/cjs/loader:1734:14)
    at Object..js (node:internal/modules/cjs/loader:1899:10)
    at Module.load (node:internal/modules/cjs/loader:1469:32)
    at Module._load (node:internal/modules/cjs/loader:1286:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    ... 2 lines matching cause stack trace ...
    at require (node:internal/modules/helpers:135:16) {
  [cause]: Error: Cannot find module '@rollup/rollup-darwin-arm64'
  Require stack:
  - /Users/username/rimmel/node_modules/rollup/dist/native.js
  - /Users/username/rimmel/node_modules/rollup/dist/shared/parseAst.js
  - /Users/username/rimmel/node_modules/rollup/dist/shared/rollup.js
  - /Users/username/rimmel/node_modules/rollup/dist/bin/rollup
      at Module._resolveFilename (node:internal/modules/cjs/loader:1405:15)
      at defaultResolveImpl (node:internal/modules/cjs/loader:1061:19)
      at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1066:22)
      at Module._load (node:internal/modules/cjs/loader:1215:37)
      at TracingChannel.traceSync (node:diagnostics_channel:322:14)
      at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
      at Module.require (node:internal/modules/cjs/loader:1491:12)
      at require (node:internal/modules/helpers:135:16)
      at requireWithFriendlyError (/Users/username/rimmel/node_modules/rollup/dist/native.js:41:10)
      at Object.<anonymous> (/Users/username/rimmel/node_modules/rollup/dist/native.js:68:76) {
    code: 'MODULE_NOT_FOUND',
    requireStack: [
      '/Users/username/rimmel/node_modules/rollup/dist/native.js',
      '/Users/username/rimmel/node_modules/rollup/dist/shared/parseAst.js',
      '/Users/username/rimmel/node_modules/rollup/dist/shared/rollup.js',
      '/Users/username/rimmel/node_modules/rollup/dist/bin/rollup'
    ]
  }
}
```

Node.js v24


A workaround seems to be manually installing a package:

```sh
npm i @rollup/rollup-darwin-arm64
```

A long-term fix is required instead, possibly not by just adding it to the package.json, as it would install it for non-Mac users, as well.

--- a/.npmrc
+++ b/.npmrc
@@ -1,0 +1,4 @@
+# Fix for npm optional dependencies bug affecting Rollup on macOS
+# See: https://github.com/npm/cli/issues/4828
+# This file ensures optional dependencies are properly installed
+

--- a/README.md
+++ b/README.md
@@ -798,6 +798,16 @@ npm run test
 npm run build
 ```
 
+### macOS Users
+If you encounter an error about missing `@rollup/rollup-darwin-arm64` module, this is due to an npm bug with optional dependencies. To fix:
+
+```bash
+rm -rf node_modules package-lock.json
+npm install
+```
+
+The `.npmrc` file in the repository helps prevent this issue.
+
 There is a "kitchen sink" app you can use to play around locally, which should showcase most of what you can do with Rimmel:
 ```bash
 npm run kitchen-sink

